packages/components/tests/integration/components/hds/table/th-sort-test.js

```diff
@@ -14,31 +14,44 @@ module('Integration | Component | hds/table/th-sort', function (hooks) {
 
   test('it renders', async function (assert) {
     await render(hbs`<Hds::Table::ThSort
-  @sortBy='artist'
-  @sortOrder='asc'
   @setSortBy={{this.setSortBy}}
 >artist</Hds::Table::ThSort>`);
 
     assert.dom(this.element).hasText('artist');
   });
+  test('it has the scope attribute, and it is set to column', async function (assert) {
+    await render(hbs`<Hds::Table::ThSort
+  @setSortBy={{this.setSortBy}}
```

Comment:
[blocker] if I understand correctly, this is not actually saving you from repetition (and also makes the code more obscure/less explicit)

Reply:
This test is included because it is a clue that something is missing if the attribute `scope="col"` is ever missing. It can by all means be removed but the test is an indicator that it is put there purposefully and should not be removed.

Reply:
Also, can you clarify what you consider a blocker here?

Reply:
I think the code has been refactored since my comment (or github assigned the comment to the wrong block of code). My comment was about this code: 
```
  // since we'd have to repeat this for each test, we'll set it up once to be used in all tests
  hooks.beforeEach(function () {
    this.set('setSortBy', function (sortBy) {
      return sortBy || 'artist';
    });
  });
````

Reply:
The tests fail if I take it out: 
<img width="1359" alt="CleanShot 2022-10-25 at 19 45 21@2x" src="https://user-images.githubusercontent.com/4587451/197908429-71257fa6-5526-473e-9b9a-c7fe5883bfca.png">

Reply:
What I mean is this. You can have much a simpler set of tests like this:

```
module('Integration | Component | hds/table/th-sort', function (hooks) {
  setupRenderingTest(hooks);

  test('it renders', async function (assert) {
    await render(hbs`<Hds::Table::ThSort>Artist</Hds::Table::ThSort>`);
    assert.dom(this.element).hasText('Artist');
  });
  test('it has the scope attribute, and it is set to column', async function (assert) {
    await render(hbs`<Hds::Table::ThSort>Artist</Hds::Table::ThSort>`);
    assert.dom('[data-test-table-th-sort]').hasAttribute('scope', 'col');
  });

  test('it should render the component with a CSS class that matches the component name', async function (assert) {
    await render(hbs`<Hds::Table::ThSort>Artist</Hds::Table::ThSort>`);
    assert.dom('[data-test-table-th-sort]').hasClass('hds-table__th-sort');
  });

  test('if @sortOrder is not defined, the swap-vertical icon should be displayed', async function (assert) {
    await render(
      hbs`<Hds::Table::ThSort @sortBy='artist'>artist</Hds::Table::ThSort>`
    );
    assert.dom('[data-test-icon="swap-vertical"]').exists();
  });

  test('if `@sortOrder` is set to asc, the arrow-up icon should be displayed', async function (assert) {
    await render(
      hbs`<Hds::Table::ThSort @sortBy='artist' @sortKey='artist' @sortOrder='asc'>Artist</Hds::Table::ThSort>`
    );
    assert.dom('[data-test-icon="arrow-up"]').exists();
  });
});
```
where all the code related to `setSortBy` is completely removed, and to do that we need to change how the `setSortBy` is invoked. 

Currently is invoked like this:
```
<button type="button" {{on "click" (fn @setSortBy @sortKey)}}>
```
but this triggers an error if `@setSortBy` is not set. Instead we generally use a "guard" to avoid this error (you can see it applied in many components: `Alert`, `Breadcrumb`, `Disclosure`, `Toggle Button`, `Form Error + Helper Text`, `Tag`). Something like this (in `th-sort.js`):
```
const NOOP = () => {};

export default class HdsTableThSortComponent extends Component {
   ...

  /**
   * @param onClick
   * @type {function}
   * @default () => {}
   */
  get onClick() {
    let { setSortBy, sortKey } = this.args;

    // notice: this is a guard used in case the component is used as standalone element
    if (typeof setSortBy === 'function') {
      return () => {
        setSortBy(sortKey);
      };
    } else {
      return NOOP;
    }
  }
}
```

Now the invocation in the template is like this:
```
<button type="button" {{on "click" this.onClick}}>
```
and the `@setSortBy` argument is not a required argument anymore for the `ThSort` (sub) component (when used inside the `Table` is controlled and provided by the table itself) and this makes `ThSort` simpler and avoids the need to define it in the tests as argument for the component.

PS: if you are not convinced yet, do this experiment: go in the test file `table/th-sort-test.js` and replace this code:
```
  // since we'd have to repeat this for each test, we'll set it up once to be used in all tests
  hooks.beforeEach(function () {
    this.set('setSortBy', function (sortBy) {
      return sortBy || 'artist';
    });
  });
```
with this one (essentially **`setSortBy` becomes a completely empty function**):
```
  // since we'd have to repeat this for each test, we'll set it up once to be used in all tests
  hooks.beforeEach(function () {
    this.set('setSortBy', function () {});
  });
```
and run the tests: you will see that **they pass anyway**. This means the `setSortBy` defined in the tests is not needed by the component per se, it's just there to avoid to trigger an error when the compilation occurs if it's not defined (a check in the `fn` helper, I imagine).

Reply:
So this is more about adding `NOOP` to the component in order to simplify the tests. Okay, got it. I'll look into it, ty for the feedback.

Reply:
Ok after looking into it some more (I updated the `beforeEach` hook to use optional chaining and I see what you mean), I sort of agree but also am not very convinced that the th-sort component should be made generic, since it's entire explicit purpose is to create a sortable component. 

I will think about it some more tonight and come back to it tomorrow morning.

Reply:
The fact that the `ThSort` component can live without the `setSortBy` for me is telling us that this component should know anything about sorting. If you think about it the component does not "sort", but on "click" invokes a function (provided from outside) without even knowing what that function does. The sorting logic happens inside the `Table`, not inside the `ThSort` (it's the name that leads to think that this "sorts").

Reply:
ok I've refactored the th-sort component in a separate PR. #666